core/services/feeds/service.go

```diff
@@ -406,6 +407,45 @@ func (s *service) ListJobProposalsByManagersIDs(ids []int64) ([]JobProposal, err
 	return s.orm.ListJobProposalsByManagersIDs(ids)
 }
 
+// DeleteJobArgs are the arguments to provide to the DeleteJob method.
+type DeleteJobArgs struct {
+	FeedsManagerID int64
+	RemoteUUID     uuid.UUID
+}
+
+// DeleteProposal deletes a job proposal if it exist. The feeds manager id check
+// ensures that only the intended feed manager can make this request.
+func (s *service) DeleteProposal(ctx context.Context, args *DeleteJobArgs) (int64, error) {
+	proposal, err := s.orm.GetJobProposalByRemoteUUID(args.RemoteUUID)
+	if err != nil {
+		if !errors.Is(err, sql.ErrNoRows) {
+			return 0, errors.Wrap(err, "failed to check existence of job proposal")
+		}
+	}
+
+	// Validation for existing job proposals
+	if err == nil {
+		// Ensure that if the job proposal exists, that it belongs to the feeds
+		// manager which previously proposed a job using the remote UUID.
+		if args.FeedsManagerID != proposal.FeedsManagerID {
+			return 0, errors.New("cannot delete a job proposal belonging to another feeds manager")
+		}
+	}
+
+	pctx := pg.WithParentCtx(ctx)
+	if txerr := s.orm.DeleteProposal(proposal.ID, pctx); txerr != nil {
+		s.lggr.Errorw("Failed to delete the proposal", "error", txerr)
+
+		return 0, errors.Wrap(txerr, "DeleteProposal failed")
+	}
```

Comment:
You can just use `err` here

 ```suggestion
	if err = s.orm.DeleteProposal(proposal.ID, pctx); err != nil {
		s.lggr.Errorw("Failed to delete the proposal", "error", err)

		return 0, errors.Wrap(err, "DeleteProposal failed")
	}
```